build/helper/codegen_helper.py

```diff
@@ -221,6 +222,10 @@ def _get_ctype_variable_definition_snippet_for_string(parameter, parameters, ivi
             definition = 'ctypes.create_string_buffer(self._repeated_capability.encode(self._encoding))  # case C010'
         elif parameter['enum'] is not None:
             definition = 'ctypes.create_string_buffer({0}.value.encode(self._encoding))  # case C030'.format(parameter['python_name'])
+        # _init_function is a special case - Converters should not be called within it since they are
```

Comment:
These special cases (`method_name is not config['functions']['_init_function']['python_name']`) have a really high code-smell. It also seems to be the only reason why you had to pipe down `method_name` all the way here.

We should try to come up with a cleaner alternative.

@texasaggie97 what do you think?

Reply:
Instead of making a special case here (and having to plumb the method name all the way down) why not just remove the converter from `functions['_init_function']`? It is a copy of whichever init is listed in config: https://github.com/ni/nimi-python/blob/bcf6d2204bb1821ef398c1f91685a0e0294bb808/build/helper/metadata_add_all.py#L317

Reply:
> why not just remove the converter from functions['_init_function']?

I can't do that because it is used to generate the code that calls the converters if required, in `Session.__init__`:
https://github.com/ni/nimi-python/blob/bcf6d2204bb1821ef398c1f91685a0e0294bb808/build/templates/session.py.mako#L263

Converters need to be called here to get string representations of parameters, for use in `__repr__`.